### ISSUE TITLE
import not working with typescript

### ISSUE THREAD

**Issue by gregveres** [highcharts/highcharts#7533]
I have looked at the gist that shows to use:
`import * as Highcharts from 'highcharts'`

but that isn't working for me. 

I am using typescript v2.6.2
I am using webpack for bundling with a vendor chuck that includes:

- jquery

- knockout

- bootstrap

- moment

I have a page with its own chuck that is generated from a typescript file that has the import at the top of the file. 
```
import $ = require("jquery");
import ko = require("knockout");
import * as Highcharts from 'highcharts';
```

I know the vendor chuck is getting created properly and the rest of my app can access jquery. 
I also expose jquery globally with this as part of my webpack config:

```
    plugins: [
        new webpack.ProvidePlugin({
            $: "jquery",
            jQuery: "jquery",
            "window.jQuery": "jquery"
        }),
]
```

When I go to use Highcharts, from the import statement, I get that Highcharts is not defined. If I look at it in the chrome debugger, I see a structure defined like this:

message: "Highcharts is not defined"

If I look at the closure that contains ko and $, there is no variable named highcharts. 

I assume I am doing something wrong with the import but I can't figure out what. 
I know that highcharts.js is being included. 
I know that I am defining it in the way that the docs say I should.
Maybe $ isn't defined at the time that the highcharts code is loaded?

**Comment by pawelfus** [comment]
Hi @gregveres 

Thank you for reporting the issue. Have you checked #4994 issue? The error suggests that you call Highcharts before lib is loaded.

**Comment by gregveres** [comment]
Yes I looked through #4994, that is what led me here. 
by putting break points in the beginning of the highcharts module, I can see that highcharts is indeed getting loaded. But the import * as Highcharts from "highcharts" isn't setting the variable Highcharts to the export of the module. 

When I stop in the debugger where webpack is loading the module, I see that it creates the module.exports and sets that to the highcharts object, but then it doesn't pass it along to the Typescript module that does the import *. 

do I need a ProvidePlugin call that exports highcharts globally?

**Comment by gregveres** [comment]
The strange thing is that it doesn't seem like typescript / webpack thinks I am importing highcharts. 

I have tried with highcharts in its own chunk file - and I see that the chunk file gets created as a 3.5MB file (I have substituted the source file in for the regular minified file while trying to debug this issue).

But then when I get rid of the dedicated highcharts chunk, I would expect the chunk file for my module to go up by 3.5MB. It doesn't. its size doesn't change.

**Comment by gregveres** [comment]
I can get webpack to include highcharts in the module's chunk file if I do this:

```
import $ = require("jquery");
import ko = require("knockout");
import "highcharts";
import * as Highcharts from 'highcharts';
```

but this leaves Highcharts undefined. (even though the debugger indicates that highcharts has indeed been loaded.

and of course if I replace both highcharts imports with just a single line:

`import Highcharts from 'highcharts';`

I get the error: TS1192: Module '"highcharts"' has no default export.

**Comment by gregveres** [comment]
Ok looking at the javascript file that the typescript compiler produces, there is no reference to highcharts. So that would be why the Highcharts variable is undefined. 

typescript and my configuration just don't seem to like the import * syntax. My tsconfig.json is:

```
{
  "compilerOptions": {
    "noImplicitAny": true,
    "noEmitOnError": true,
    "removeComments": true,
    "sourceMap": true,
    "target": "es5",
    "module": "commonjs",
    "moduleResolution": "node",
    "skipLibCheck": true,
    "typeRoots": [
      "../node_modules/@types",
      "typings"
    ],
    "baseUrl": "Scripts/"
  },
  "compileOnSave": true,
  "exclude": [
    "node_modules",
    "obj",
    "bin"
  ]
}

```

I will also post something on the typescript github project because I think the problem is typescript.

**Comment by gregveres** [comment]
OK I think I found my problem. 

My project is an MVC app that I am now converting to webpack and to npm for the client. But that meant that I had the definitely typed package for highcharts installed as a NuGet package. But I had installed highcharts as a node_module. This confused the typescript compiler to the point where it just  ignored the import * charts as "highcharts" line. I verified this by looking at the compiled javascript output. 

Once I removed the NuGet package and installed the npm typings file with npm @typings/highchart -D, the typescript compiler stopped being confused and started emitting the correct import syntax. 

I am now getting highcharts inserted into the module's chunk file properly and it is getting imported properly. I just have to figure out how to convert the code that used to use highcharts as a jQuery plugin to code that references the imported highcharts object.
